call-home/src/common/iterator.rs

```diff
@@ -0,0 +1,69 @@
+#[derive(Debug, Clone)]
+pub enum SortOrder {
+    NonDescending,
+    NonAscending,
+}
+
+#[derive(Debug, Clone)]
+pub struct SortedExactSizeIter<T> {
+    data: Vec<T>,
```

Comment:
This is taking ownership so again more copies?

Reply:
Yes.. I'll have to get around this one. For this application, sorting the volumes and pools should be fine, instead of collecting their sizes into separate iterators.